FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,276 @@
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = ViewGroup.LayoutParams.WRAP_CONTENT
+            gravity = Gravity.END
+        }
+
+        // Dismiss gradient gap
+        val gradientGap = View(context).apply {
+            background = ContextCompat.getDrawable(
+                    context,
+                    R.drawable.contextual_command_bar_dismiss_button_gap_background
+            )
+        }
+        dismissButtonContainer!!.addView(gradientGap)
+        gradientGap.layoutParams.apply {
+            width = resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_dismiss_gap_width)
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+        }
+
+        // Dismiss button
+        itemPaddingVertical = resources.getDimensionPixelSize(
+                R.dimen.fluentui_contextual_command_bar_default_item_padding_vertical
+        )
+        itemPaddingHorizontal = resources.getDimensionPixelSize(
+                R.dimen.fluentui_contextual_command_bar_default_item_padding_horizontal
+        )
+        dismissButton = ImageView(context).apply {
+            setPaddingRelative(
+                    itemPaddingHorizontal,
+                    itemPaddingVertical,
+                    itemPaddingHorizontal,
+                    itemPaddingVertical
+            )
+            setBackgroundColor(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissBackgroundColor
+            ))
+            imageTintList = ColorStateList.valueOf(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissIconTintColor
+            ))
+
+            setOnClickListener {
+                dismissListener?.invoke()
+            }
+        }
+        dismissButtonContainer!!.addView(dismissButton)
+        dismissButton!!.layoutParams.apply {
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            )
+        }
+
+        bringChildToFront(dismissButtonContainer)
+    }
+
+    private fun setDismissButtonVisible(visible: Boolean) {
+        if (dismissIcon == 0) {
+            // Skip if there is no icon
+            return
+        }
+        if (visible && dismissButton == null) {
+            addDismissButton()
+        }
+
+        val paddingEnd = if (visible) {
+            resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            ) + resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_gap_width
+            )
+        } else 0
+        commandContainer.setPaddingRelative(0, 0, paddingEnd, 0)
+        dismissButtonContainer?.isVisible = visible
+    }
+
+    fun setItemGroups(itemGroups: List<CommandItemGroup>) {
```

Comment:
Updating the commands has to be done in a batch operation? We don't expect more granular add/remove icons? I understand the logic to do so would be complex, but wanted to know if it was considered in the design (I think the current design assumes a relatively static set of commands)